src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableInterlocked.cs

```diff
@@ -394,17 +394,17 @@ public static bool InterlockedInitialize<T>(ref ImmutableArray<T> location, Immu
                 if (priorCollection.TryGetValue(key, out oldValue!))
                 {
                     newValue = updateValueFactory(key, oldValue);
-                    if (priorCollection.ValueComparer.Equals(oldValue, newValue))
-                    {
-                        return oldValue;
-                    }
                 }
                 else
                 {
                     newValue = addValue;
                 }
 
                 var updatedCollection = priorCollection.SetItem(key, newValue);
+				if (object.ReferenceEquals(priorCollection, updatedCollection))
+				{
```

Comment:
whitespace issues seem to be failing the build.

Reply:
Already fixed, this should be a 'Outdated' file change.